src/argilla/server/apis/v1/handlers/datasets.py

```diff
@@ -285,6 +295,67 @@ async def create_dataset_records(
         raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail=str(err))
 
 
+@router.post(
+    "/me/datasets/{dataset_id}/records/search",
+    status_code=status.HTTP_200_OK,
+    response_model=SearchRecordsResult,
+    response_model_exclude_unset=True,
+)
+async def search_dataset_records(
+    *,
+    db: Session = Depends(get_db),
+    search_engine: SearchEngine = Depends(get_search_engine),
+    telemetry_client: TelemetryClient = Depends(get_telemetry_client),
+    dataset_id: UUID,
+    query: SearchEngineQuery,
```

Comment:
Maybe this could be changed and having a specific schema model for the API call. Also, once we start introducing complex filters (metadata combinations, with other things), they should be included as part of the body. So, even if can be redundant no, maybe makes sense to define the endpoint body as:

```json
{
   "query": { "text": ...}
}
```
later, adding filters in the body can be just:
```
{ 
   "query": { "text": ...}, # This part will be relevant to the score computation
   "filter": {   # This part won't be used for the score computation
      "fields": { 
         "text": { 
            "field": "category",
            "q": "animals"
         }
     }    
}
```

Reply:
agree!